cmake/OpenCVFindProtobuf.cmake

```diff
@@ -67,22 +74,51 @@ else()
   endif()
 endif()
 
+# See https://github.com/opencv/opencv/issues/24369
+# In Protocol Buffers v22.0 and later drops C++11 support and depends abseil-cpp.
+#   Details: https://protobuf.dev/news/2022-08-03/
+# And if std::text_view is in abseil-cpp requests C++17 and later.
+
+if(HAVE_PROTOBUF)
+    if("${Protobuf_VERSION}" MATCHES [[[0-9]+.([0-9]+).[0-9]+]])
+        string(COMPARE GREATER "${CMAKE_MATCH_1}" "21" REQUEST_AFTER_CXX17)  # >=22
+
+        if(REQUEST_AFTER_CXX17)
+            string(COMPARE GREATER "${CMAKE_CXX_STANDARD}" "16" USED_AFTER_CXX17)  # >=17
+
+            if(NOT USED_AFTER_CXX17)
```

Comment:
Use `if(GREATER)` instead.

Reply:
This comparing result is used with "NOT", so I use `LESS`.
And I add to check C++98. Thank you !